### DISCUSSION TITLE
Shipping details update by SP-API

### DISCUSSION THREAD

**Discussion by hardikradixweb** [amzn/selling-partner-api-models#3379] [General]
Hello,

I have integrated the getOrders API and it works fine but now I have to update the shipping details for the orders like carrier, tracking no so how can do that? I mean which API i should use?

I want to update shipping information same like this: https://sellercentral.amazon.in/gp/help/external/200259170?language=en_IN&ref=efph_200259170_cont_200418100

**Comment by mianzaid** [comment]
> Hello,
> 
> I have integrated the getOrders API and it works fine but now I have to update the shipping details for the orders like carrier, tracking no so how can do that? I mean which API i should use?
> 
> I want to update shipping information same like this: https://sellercentral.amazon.in/gp/help/external/200259170?language=en_IN&ref=efph_200259170_cont_200418100


You can use Amazon [Feed](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-reference) to update the shaping details. Before jump into the feed please read the post https://developer-docs.amazon.com/sp-api/docs/feeds-api-best-practices.

**Reply by hardikradixweb** [reply]
Hello,

I got success in creating feed document, uploading document and creating feeds to amazon.But i am facing issue with the uploaded document to the amazon and I have tried multiple ways, text/xml, text/csv, text/plain but every time i got the same error.

Error 1: Warning	The order-id\torder-item-id\tquantity\tship-date\tcarrier-code\tcarrier-name\ttracking-number\tship-method\ttransparency_code\tship_from_address_name\tship_from_address_line1\tship_from_address_line2\tship_from_address_line3\tship_from_address_city\tship_from_address_county\tship_from_address_state_or_region\tship_from_address_postalcode\tship_from_address_countrycode\t\r\n123456789\t333333\t10\t10-02-30\tFEDEX\tFedEx\t852852828\tGround\t\t\t\t\t\t\t\t\t\t\t\r\n123456789\t333333\t10\t10-02-30\tFEDEX\tFedEx\t852852828\tGround\t\t\t\t\t\t\t\t\t\t\t\r\n field heading is invalid, so values in this field were ignored. To correct this error, download the template again and use the new copy, or fix the field heading in your existing file.

Text file: [amz-file.txt](https://github.com/amzn/selling-partner-api-docs/files/9575429/amz-file.txt)

Error 2: 50001 error "The XML you submitted is ill-formed at the Amazon Envelope XML level at (or near) line 2, column 18.
XML:
![image](https://user-images.githubusercontent.com/152806/190410247-683521d4-8d7b-4980-a37e-c6f88483d167.png)


Anyone have faced any issues that feeds got created and processed with the error?
